app/api/schema.py

```diff
@@ -1,5 +1,134 @@
+
+    class Input:
+        username = graphene.String(required=True)
+        first_name = graphene.String(required=False)
+        last_name = graphene.String(required=False)
+        email = graphene.String(required=False)
+        is_staff = graphene.Boolean(required=False)
+        is_active = graphene.Boolean(required=False)
+        password = graphene.String(required=True)
+
+    user = graphene.Field(UserNode)
+    errors = graphene.List(graphene.String)
+
+    @classmethod
+    def mutate_and_get_payload(cls, input, context, info):
+        try:
+            user = User()
+            user.username = input.get('username')
+            user.first_name = input.get('first_name', '')
+            user.last_name = input.get('last_name', '')
+            user.email = input.get('email', '')
+            user.is_staff = input.get('is_staff', False)
+            user.is_active = input.get('is_active', False)
+            if input.get('password'):
+                user.set_password(input.get('password'))
+            user.full_clean()
+            user.save()
+            return CreateUser(user=user)
+        except ValidationError as e:
+            return CreateUser(user=None, errors=getErrors(e))
+
+
+class UpdateUser(graphene.relay.ClientIDMutation):
+
+    class Input:
+        id = graphene.String(required=True)
+        username = graphene.String(required=False)
+        first_name = graphene.String(required=False)
+        last_name = graphene.String(required=False)
+        email = graphene.String(required=False)
+        is_staff = graphene.Boolean(required=False)
+        is_active = graphene.Boolean(required=False)
+        password = graphene.String(required=False)
+
+    user = graphene.Field(UserNode)
+    errors = graphene.List(graphene.String)
+
+    @classmethod
+    def mutate_and_get_payload(cls, input, context, info):
+        user = get_user(input.get('id'))
+        for key, value in input.items():
+            if key == 'password':
+                user.set_password(input.get('password'))
+            elif key != 'id':
+                setattr(user, key, value)
+        try:
+            user.full_clean()
+            user.save()
+            return UpdateUser(user=user)
+        except ValidationError as e:
+            return UpdateUser(user=user, errors=getErrors(e))
+
+
+class DeleteUser(graphene.relay.ClientIDMutation):
+
+    class Input:
+        id = graphene.String(required=True)
+
+    deleted = graphene.Boolean()
+    user = graphene.Field(UserNode)
+
+    @classmethod
+    def mutate_and_get_payload(cls, input, context, info):
+        try:
+            user = get_user(input.get('id'))
+            user.delete()
+            return DeleteUser(deleted=True, user=user)
+        except:
+            return DeleteUser(deleted=False, user=None)
+
+
+class Query(graphene.AbstractType):
+    user = graphene.relay.Node.Field(UserNode)
+    users = DjangoConnectionField(UserNode)
```

Comment:
@andela-aoduola: Instead of `DjangoConnectionField`, `DjangoFilterConnectionField` from `graphene_django.filter` module should be used in order for the filters set on line 24 to work. You didn't notice it because there are no test cases for the filtering.